### ISSUE TITLE
Is there a limit on number of spans that are open at the same time?

### ISSUE THREAD

**Issue by hubertlepicki** [appsignal/appsignal-elixir#624]
I am observing weird thing that is happening on prod. I added some instrumentation with Appsignal.instrument/1, and inside I am doing some logging, including I am logging "#{Tracer.current_span()}".

This seems to be nil from time to time.

I am also pretty sure, that whenever I go to Actions -> some action -> samples I do *not* see all samples there either. Should I see all requests for that action or "samples" really means just selected samples?

The same is the question with Appsignal.instrument / Tracer.current_span - should I actually see all exceptions that are thrown inside that instrumentation or - what I see in Appsignal Dashboard web app is just sample of whatever happens on prod and it's intended that we don't see every single occurrence of exception there?

This has *never* happened on staging system, which is otherwise identical to our production system except it has obviously far smaller number of requests / load.

I suspect that it's either of:

1. Intended behavior - and "samples" really have a limit and whenever I open Span I shouldn't expect it to be there as there's some limit on how many spans can be opened
2. There is some sort of race condition where just opened span maybe is not always visible immediatelly in Tracer.current_span() ?
3. I have calls to Tracer.ignore() that and something is recycling the previously used pids. I don't think this is the case but maybe one of the libs we use has a pool of processes that are being started/recycled that were ignored previously.

**Comment by shairyar** [comment]
Hi @hubertlepicki,

Could you please share some example code demonstrating this issue?

> I am observing weird thing that is happening on prod. I added some instrumentation with Appsignal.instrument/1, and inside I am doing some logging, including I am logging "#{Tracer.current_span()}".
> 
> This seems to be nil from time to time.

Since most of these issues are support-related, could you please send those over to support@appsignal.com?

**Comment by thijsc** [comment]
HI @hubertlepicki, looks like there are a few more support type questions and possibly also a bug in this bug report. Would you mind contacting support@appsignal.com with any product questions you have? We'll look into the bug report as soon as we have a bit more information on what your code looks like.

**Comment by hubertlepicki** [comment]
I actually figured it out. Which may be something you want to fix on your end as it's super confusing.

I had nested spans, accidentally, and been calling add_error on these nested spans. This seems to do nothing. There is no error recorded in AppSignal dashboard as I guess the nested span is not an entity on it's own.

What I would like to happen is that either was true:

- whenever you add errors to nested span, it adds them to the root span instead
- the system prevents you / warns you if you do add errors that otherwise will not be reported. Logger error message would help for example, esp. if it was picked up by Appsignal integration and opened error in that case
